cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+## Requirements
+
+- Users should be able to **re-use a configuration** they have used for cluster creation previously, optionally overriding only a few details.
+
+- A **dry run option** should allow "previewing" the result without actually writing any resources to the management cluster.
+
+- A **connection with the management cluster** should only be required to write resources. For a dry run, no connection should be necessary. (However, an internet connection will be required to access public information from GitHub repositories.)
+
+- Configuration provided by the user should be **validated** against the apps's schema, and in case of validation errors, users should receive detailed feedback.
+
+- Users should decide whether to actually create the cluster, or print a manifest only.
+
+- A **cluster name must be provided by the user**. We won't generate a cluster name for the user.
+  - At Giant Swarm we decided around November 2023 that we want users to set workload cluster names and avoid auto-generated names ([roadmap#2999](https://github.com/giantswarm/roadmap/issues/2999)). In theory, we could still offer an option (a flag) to auto-generate the name. However, this would require knowledge of the cluster-PROVIDER app schema in the command logic. For example, as of cluster-aws v0.73.0, the cluster name is set in the property .global.metadata.name. Although we are working towards alignment across providers, this may be different in other versions or other provider apps.
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--app-version`: the cluster-PROVIDER app version to use. If not specified, the latest version is used.
+- `--dry-run`: only validate the configuration and print the resulting manifest.
+- `--output`: file path to write the result to. If not specified, resources are written to the management cluster. The option `STDOUT` can be used to print the result to the console.
+- `--format`: This flag can be used to specify the format of the output.
+  - Default is `manifest` for a Kubernetes manifest that includes an App resources and one or several ConfigMap resources.
+  - `config` for merged configuration YAML.
+  - `command` for a self-contained representation of the command just executed, as a one-liner.
```

Comment:
```suggestion
  - `manifest` is the default for a Kubernetes manifest that includes App resources and one or several ConfigMap resources.
  - `config` for merged configuration YAML.
  - `command` for a self-contained representation of the command just executed, as a one-liner.
```
I don't get what it means, I don't see examples below either. Can we elaborate on what it really does?

Reply:
Clarified in 2754756677d4deac32c2b3939bfe9543a44c26e7